sklearn/datasets/_samples_generator.py

```diff
@@ -1241,6 +1245,22 @@ def make_low_rank_matrix(
 
 # TODO(1.3): Change argument `data_transposed` default from True to False.
 # TODO(1.3): Deprecate data_transposed, always return data not transposed.
+
+
+@validate_params(
+    {
+        "n_samples": [Interval(Integral, 0, None, closed="neither")],
+        "n_components": [Interval(Integral, 0, None, closed="neither")],
+        "n_features": [Interval(Integral, 0, None, closed="neither")],
+        "n_nonzero_coefs": [Interval(Integral, 0, None, closed="neither")],
+        "random_state": [
+            "random_state",
+            Interval(Integral, 0, None, closed="neither"),
+            None,
+        ],
+        "data_transposed": ["boolean"],
+    }
+)
 def make_sparse_coded_signal(
     n_samples,
     *,
```

Comment:
In the description below, could you change the following:

```
    Returns a matrix `Y = DX`, such that `D` is of shape `(n_features, n_components)`,
    `X` is of shape `(n_components, n_samples)` and each column of `X` has exactly
    `n_nonzero_coefs` non-zero elements.
```

Reply:
done!